### PR TITLE
[Infineon] Fix CYW30739 KVS to pass the storage API audit.

### PR DESCRIPTION
**Pull request by hsusid**
#### Problem
Fixes #22030

https://github.com/project-chip/connectedhomeip/pull/20134 and https://github.com/project-chip/connectedhomeip/pull/20164 provide a storage API compliance audit to verify the platform KVS storage implementations.
Current CYW30739 KVS implementations fail to pass the storage API compliance audit.

#### Change overview
* Support key length up to PersistentStorageDelegate::kKeyLengthMax.
* _Get and _Put methods can handle zero-size data correctly.

#### Testing
Build with API audit enabled:
`gn gen out/debug --args="chip_error_logging=true chip_openthread_ftd=false chip_support_enable_storage_api_audit=true"`

API audit passes with the following line:
`ErrM CHIP:ATM: ==== PersistentStorageDelegate API audit: SUCCESS ====`

### CODE REVIEW

**Review by tcarmelveilleux — APPROVED** [review]

File: src/platform/CYW30739/KeyValueStoreManagerImpl.cpp [hunk 1]
```diff
@@ -75,14 +74,24 @@ CHIP_ERROR KeyValueStoreManagerImpl::_Get(const char * key, void * value, size_t
...
+    if (value_size == 0 || entry->GetValueSize() == 0)
+    {
+        if (read_bytes_size != nullptr)
+            *read_bytes_size = 0;
+
+        if (value_size >= entry->GetValueSize())
+            ExitNow(err = CHIP_NO_ERROR);
+        else
+            ExitNow(err = CHIP_ERROR_BUFFER_TOO_SMALL);
+    }
```
Comment (tcarmelveilleux) [inline src/platform/CYW30739/KeyValueStoreManagerImpl.cpp:94]: Can you please add issues in https://github.com/orgs/project-chip/projects/3 about your platform's audit and if you had issues, please raise small issues attached to that project and mark with `fixes #XXX` in the PR description. I helps understand the impact of the audit. No need to changer the lines here, just wanted to make sure this comment was seen.

Comment (hsusid) [reply src/platform/CYW30739/KeyValueStoreManagerImpl.cpp:94]: It seems I don't have access to https://github.com/orgs/project-chip/projects/3. It gave me a 404 error.
How do I apply for access to the project?

Comment (hsusid) [reply src/platform/CYW30739/KeyValueStoreManagerImpl.cpp:94]: I have created an issue for this https://github.com/project-chip/connectedhomeip/issues/22030.

**Comment by hsusid** [comment]
rebase to re-trigger CI

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**